chunkset_tpl.h

```diff
@@ -37,7 +37,8 @@ Z_INTERNAL uint8_t* CHUNKCOPY(uint8_t *out, uint8_t const *from, unsigned len) {
 
 /* Behave like chunkcopy, but avoid writing beyond of legal output. */
 Z_INTERNAL uint8_t* CHUNKCOPY_SAFE(uint8_t *out, uint8_t const *from, unsigned len, uint8_t *safe) {
-    if ((safe - out) < (ptrdiff_t)sizeof(chunk_t)) {
+    len = MIN(len, safe - out + 1);
+    if (len < (ptrdiff_t)sizeof(chunk_t)) {
```

Comment:
I don't think we need `(ptrdiff_t)` cast here.

Reply:
`sizeof` should return `size_t` which is unsigned type... on the other hand, pointer math with no explicit cast will return `int` which is signed type. Unlike the standard says, modern compilers will actually cast both sides of comparison to same type if one side uses explicit cast as long as signedness match.